app/telemetry.go

```diff
@@ -67,43 +66,16 @@ func (a *App) StoreTelemetryData(dItm *telemetrylib.TelemetryDataItem, bHdr *tel
 	return
 }
 
-type TelemetryDataRow interface {
+type TelemetryDataRowHandler interface {
+	// TelemetryDataRow is a superset of TableRow
+	TableRowHandler
+
 	// Initialise the row fields
 	Init(dItm *telemetrylib.TelemetryDataItem, bHdr *telemetrylib.TelemetryBundleHeader, tagSetId int64) error
-
-	// Setup DB access
-	SetupDB(db *sql.DB) error
-
-	// Retrieve the TableName
-	TableName() string
-
-	// Retrieve the RowId
-	RowId() int64
-
-	// Return string representation of the row
-	String() string
-
-	// Check if the row exists in the DB, and if so populate it
-	Exists() bool
-
-	// Insert row into the table
-	Insert() error
-
-	// Update row in the table
-	Update() error
-
-	// Delete row from the table
-	Delete() error
 }
 
 type TelemetryDataCommon struct {
```

Comment:
Do we still need TelemetryDataCommon with these changes?

Reply:
Yes for now as the TelemetryDataCommon covers the set of fields common to all telemetry data table rows.
The Default and SccHwInfo telemetry data row types are supersets of telemetry data common. That will probably change somewhat in the long run, but for now we need the common subset.